performancecopilot/pcp
# pmlogger_daily -k 0 fixes

Changes committed to git@github.com:kmcdonell/pcp.git 20200706

Ken McDonell (4):
      qa/119: re-order setup
      pmlogger_daily: -k 0 fixes
      qa/1232: (new) check pmlogger_daily -k 0
      qa: assorted changes/remakes to match changed pmlogger_daily -k semantics

 man/man1/pmlogger_check.1      |  168 ++++++++++++++++++++--------
 qa/119                         |   30 ++---
 qa/1232                        |  107 +++++++++++++++++
 qa/1232.out                    |   36 ++++++
 qa/151                         |    6 -
 qa/151.out                     |    2 
 qa/338                         |    6 -
 qa/338.out                     |    6 -
 qa/530                         |    6 -
 qa/530.out                     |    6 -
 qa/532                         |    4 
 qa/532.out                     |    2 
 qa/679.out                     |   26 ++--
 qa/793.out                     |    6 -
 qa/925.out                     |    2 
 qa/group                       |    1 
 src/pmlogger/pmlogger_daily.sh |  245 +++++++++++++++++++++++++++++++----------
 17 files changed, 509 insertions(+), 150 deletions(-)

Details ...

commit 4bb6a878e7a6458b2ebfc6bd099a77432cfc19b4
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Tue Jul 7 12:57:07 2020 +1000

    qa: assorted changes/remakes to match changed pmlogger_daily -k semantics
    
    Since -k now has robust semantics relative to "now", not some midnight
    in the past or future, we need to tweak some of these tests a little.

commit ee9e6291e996e30533f7c369ee6b2c03cd8cffa1
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Tue Jul 7 12:56:36 2020 +1000

    qa/1232: (new) check pmlogger_daily -k 0

commit 27a818f9937a9c074a339f63e0ff36b849f5cd0e
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Tue Jul 7 08:22:26 2020 +1000

    pmlogger_daily: -k 0 fixes
    
    This commit completes the work started a while ago, but interrupted
    by the "signal pmlogger to rotate logs" work, the wrestling with systemd
    and the pmlogctl development.
    
    Included in this commit is:
    
    - consistent use of stderr for all verbose, very verbose, warning and
      error output
    - complete implementation of the extended DD[:MM[:HH]] notation for the
      -x and -k options to allow specification of "periods" other than an
      integral number of days, and in particular less than 1 day
    - integration of the pmlogctl #!# protcol in the control files for instances
      stopped by pmlogctl
    - removing redundant "pmlogger_check:" preamble for some messages to reduce
      output clutter with -V and -VV
    - proper handling of -k 0 (or indeed any period less than one day)
    - change in -k semantics to match the recent changes to -x semantics, namely
      the "period" is measured from _now_, not some arbitrary midnight as chosen
      by find(1) ... this removes a cross-platform inconsistency that has been
      there since day one

commit 8e7ebe9ab2ae70ce69ee22f0ebaf2912ff5c1baf
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Mon Jul 6 15:43:57 2020 +1000

    qa/119: re-order setup
    
    Need to stop current pmlogger BEFORE dinking with
    /etc/pcp/pmlogger/control.d
